src/librustc_mir/build/scope.rs

```diff
@@ -141,6 +144,9 @@ pub struct Scope<'tcx> {
 
     /// The cache for drop chain on “normal” exit into a particular BasicBlock.
     cached_exits: FxHashMap<(BasicBlock, CodeExtent), BasicBlock>,
+
+    /// The cache for drop chain on "generator drop" exit.
+    cached_generator_drop: Option<BasicBlock>,
```

Comment:
There's is 3 ways to exit scopes and all of them generator different code. I do think it would be a good idea to generate them with the same code. However I'm not volunteering to write that...